### PR TITLE
[BT-685]: Rename Sorter.simpleSort to sortBy.

### PR DESCRIPTION
**Pull request by cjcenizal**
- Improve sort method to accept a property name as a value provider.

### CODE REVIEW

**Comment by jenslonkowski** [comment]
I think putting all the logic inside one method is too hard to follow. I would prefer having smaller methods that each hold small portions of the logic:
- `sort` only accepts an array of valueProviders
- `simpleSort` only accepts a single valueProvider and puts it in an array to call `sort`
- `sortBy` only accepts a string and creates a valueProvider to call `simpleSort`

**Comment by ahollesen** [comment]
I think Jens' suggestion would make it slightly easier to use, so I would prefer, but I'd be fine with the current state, so 👍

**Comment by cjcenizal** [comment]
@jenslonkowski Great point about the code being difficult to follow. I think your suggestion means reducing functionality though (because you couldn't use `sort` to sort against a property name). I'm also concerned because the interface because more complex -- it's hard to tell the functions apart by their names.

Conceptually, I think we only need two functions: one "master" sorting function (which will mostly be used for sorting along multiple axes, e.g. sorting a table by different columns), and one convenience function for sorting along a single axis (e.g. sort an array of items by name).

I think the code is difficult to follow because of the interrelated (yet spatially separate) conditionals within the sort function. I've refactored the code to use a single switch statement instead of multiple conditional statements. I've also abstracted the concept of a "value provider", which should reduce the mental overhead when reading the sort function logic.

File: src/framework/services/sort/Sorter.spec.js [hunk 1]
```diff
@@ -244,13 +267,13 @@ describe('Sorter', () => {
...
-              a: item => item.a,
-              b: item => item.b,
+          it('defines a provider via a key', () => {
+            const valueProvidersObject = {
+              provider1: item => item.a,
+              provider2: item => item.b,
             };
             const sortedItems =
-              Sorter.sort(items, valueProvidersByProperty, 'b');
+              Sorter.sort(items, valueProvidersObject, 'provider2');
```
Comment (cjcenizal) [inline src/framework/services/sort/Sorter.spec.js]: Just renamed variables in this test to make it clearer.

File: src/framework/services/sort/Sorter.spec.js [hunk 2]
```diff
@@ -137,7 +160,7 @@ describe('Sorter', () => {
       describe('isDescending parameter', () => {
         describe('when false (default)', () => {
           it('sorts items ascending', () => {
-            const sortedItems = Sorter.sort(items, valueProvidersByIndex, 0);
+            const sortedItems = Sorter.sort(items, valueProvidersArray, 0);
```
Comment (cjcenizal) [inline src/framework/services/sort/Sorter.spec.js]: Renamed for clarity.

File: src/framework/services/sort/Sorter.spec.js [hunk 3]
```diff
@@ -63,39 +63,39 @@ describe('Sorter', () => {
...
-        Sorter.simpleSort(items, valueProvider);
+        Sorter.sortBy(items, valueProvider);
         expect(Sorter.sort)
           .toHaveBeenCalledWith(items, valueProvider, undefined, undefined);
       });
     });
 
     describe('sort method', () => {
-      const valueProvidersByIndex = [
+      const valueProvidersArray = [
```
Comment (cjcenizal) [inline src/framework/services/sort/Sorter.spec.js]: Renamed for clarity.

File: src/framework/services/sort/Sorter.js [hunk 4]
```diff
@@ -1,5 +1,60 @@
...
+      throw new Error(
+        `providerKeyOrIndex must be a string or number.
+        Got ${providerKeyOrIndex}.`
+      );
+    }
+  }
+
+  getValueFor(item) {
+    // NOTE: For now (for the sake of simplicity), we're assuming the providers
+    // are functions, not property names.
```
Comment (ahollesen) [inline src/framework/services/sort/Sorter.js]: This could be checked inside the constructor as well.

**Comment by ahollesen** [comment]
One small comment, else 👍

**Comment by jenslonkowski** [comment]
Great improvements! 👍
